src/main/java/com/atlassian/db/replica/internal/ReplicaPreparedStatement.java

```diff
@@ -81,25 +83,25 @@ protected ReplicaPreparedStatement(
     @Override
     public ResultSet executeQuery() throws SQLException {
         checkClosed();
-        final RouteDecisionBuilder decisionBuilder = new RouteDecisionBuilder(Reason.READ_OPERATION).sql(sql);
-        final PreparedStatement statement = getReadStatement(decisionBuilder);
-        return execute(statement::executeQuery, decisionBuilder.build());
```

Comment:
Ok so tests caught this difference.
Earlier the `decisionBuilder` was mutated in `getReadStatement` and the changes were reflected here.
Now the mutations are hidden.
Thinking...

Reply:
This *could* be fixed, but it's a wide change. I'll put this "on hold" as a draft